client/mysql.cc

```diff
@@ -5031,14 +5032,9 @@ void tee_putc(int c, FILE *file)
 #endif
 #endif
 
-static ulong start_timer(void)
+static ulonglong start_timer(void)
 {
-#if defined(__WIN__)
-  return clock();
-#else
-  struct tms tms_tmp;
-  return times(&tms_tmp);
-#endif
+  return my_timer_microseconds();
```

Comment:
Internally it uses ```gettimeofday()```, which is (according to man) affected by "discontinuous jumps in the system time".

I'd suggest to change it to ```microsecond_interval_timer()```, which uses ```clock_gettime()```. It's not perfect, but is less affected by these jumps (and we may fix it).

Also you won't have to include my_rdtsc.h in this case.